util/ranger/types.go

```diff
@@ -216,6 +224,20 @@ func (ran *Range) PrefixEqualLen(sc *stmtctx.StatementContext) (int, error) {
 	return len(ran.LowVal), nil
 }
 
+const EmptyRangeSize = int64(unsafe.Sizeof(Range{}))
+
+func (ran *Range) MemUsage() (sum int64) {
+	sum = EmptyRangeSize + int64(cap(ran.LowVal))*types.EmptyDatumSize + int64(cap(ran.HighVal))*types.EmptyDatumSize + int64(cap(ran.Collators))*16
```

Comment:
The memory usage for the slice is depend on the `cap` not `length`?

Reply:
How to understand the constant number `16` for `int64(cap(ran.Collators))*16`?

Reply:
> The memory usage for the slice is depended on the `cap` not `length`?

Yes. We need to consider the pre-allocated but unused memory.

Reply:
When we estimate the mem usage of datum, we also use `cap`:
https://github.com/pingcap/tidb/blob/3f0cea3206feb6b57e6a966299362efe3023cbef/types/datum.go#L2025-L2029

Reply:
> How to understand the constant number `16` for `int64(cap(ran.Collators))*16`?

`16` is the size of `Collator`, which is an interface. Interface is implemented by two points, one points to the type and the other points to the object. Hence the size of interface is 16. ref https://syslog.ravelin.com/what-is-an-interface-ee67b3cc9e97

Reply:
Get it. Please add the comment to explain it.

Reply:
Done.